tests/src/base/test_forcing.py

```diff
@@ -270,3 +276,54 @@ def recipe_output_cls(cls, *args, **kwargs):
             .isnull()
             .any("time")
         )
+
+
+@pytest.fixture
```

Comment:
I don't think this mock is not being used below. Have a look at the test_container.py file.

The fixture needs to be an input argument for the test function.

By adding `mocked_thing.assert_called_once_with(...` you can make sure that the mock was correctly used.

Reply:
Ah, will look into this

Reply:
It indeed wasn't actually mocking it properly using  `with mock.patch("xarray.open_dataset")`
If i use `with mock.patch("xarray.open_dataset")...`: it calls it correctly and opens the dataset, but also mocks `open_dataset` in the `caravan_forcing.to_xarray()` which then break things...
Will keep trying to fix it

Reply:
ah fixed it, https://changhsinlee.com/pytest-mock/ this is a useful resource